### PR TITLE
drivers:adc:ad400x: Added macro USE_STANDARD_SPI

### PR DESCRIPTION
**Pull request by naveengs-adi**
SPI engine used in this driver is FGPA specific. Hence added a
macro to switch between using standard SPI for specifically with
MCU.

Signed-off-by: Naveen Godekeresiddaramaiah <Naveen.Godekeresiddaramaiah@analog.com>

### CODE REVIEW

File: drivers/adc/ad400x/ad400x.h [hunk 1]
```diff
@@ -40,7 +40,13 @@
 #ifndef SRC_AD400X_H_
 #define SRC_AD400X_H_
 
+#include <stdbool.h>
```
Comment (amiclaus) [inline drivers/adc/ad400x/ad400x.h:43]: is this include related to the commit description/subject?

Comment (naveengs-adi) [reply drivers/adc/ad400x/ad400x.h:43]: Yes, It is required.
* In ad400x.h, the ad400x_init_param struct contains few members of type bool.
* When spi_engine.h is included in ad400x.h, it internally includes spi_engine_private.h and it contains <stdbool.h>
* But, When we add compile time switch USE_STANDARD_SPI, it is required to include <stdbool.h> in header file ad400x.h.

**Review by buha — APPROVED** [review]
Looks good to me, although i think it'd be more intuitive to not use negation every time:
```
#if ! USE_STANDARD_SPI
    spi_engine...()
#endif
```

It could as well be:
```
#if USE_SPI_ENGINE
    spi_engine...()
#endif
```

I'm approving as-is, it's up to you to choose one or the other.

**Comment by naveengs-adi** [comment]
> Looks good to me, although i think it'd be more intuitive to not use negation every time:
@buha Yes I agree with the statement.

I think we are good to go as is, because EVAL-AD400xFMCZ should support SPI Engine by default and the macro USE_STANDARD_SPI is the compile time switch that should be validated to check if we want to support these drivers usage with MCUs.

**Review by amiclaus — APPROVED** [review]
